data-prepper-api/src/test/java/com/amazon/dataprepper/model/configuration/PluginSettingsTests.java

```diff
@@ -3,79 +3,160 @@
+    private static final String TEST_STRING_DEFAULT_VALUE = "DEFAULT";
+
+    private static final int TEST_INT_DEFAULT_VALUE = 1000;
+    private static final int TEST_INT_VALUE = TEST_INT_DEFAULT_VALUE + 1;
+
+    private static final boolean TEST_BOOL_DEFAULT_VALUE = Boolean.FALSE;
+    private static final boolean TEST_BOOL_VALUE = !TEST_BOOL_DEFAULT_VALUE;
+
+    private static final long TEST_LONG_DEFAULT_VALUE = 1000L;
+    private static final long TEST_LONG_VALUE = TEST_LONG_DEFAULT_VALUE + 1;
+
+    private static final String TEST_INT_ATTRIBUTE = "int-attribute";
+    private static final String TEST_STRING_ATTRIBUTE = "string-attribute";
+    private static final String TEST_BOOL_ATTRIBUTE = "bool-attribute";
+    private static final String TEST_LONG_ATTRIBUTE = "long-attribute";
+
 
     @Test
-    public void testNullSettingsInPluginSetting() {
-        final PluginSetting pluginSetting = new PluginSetting(TEST_PLUGIN_NAME, null);
+    public void testPluginSetting() {
+        final String TEST_PIPELINE = "test-pipeline";
+        final int TEST_WORKERS = 1;
+
+        final PluginSetting pluginSetting = new PluginSetting(TEST_PLUGIN_NAME, ImmutableMap.of());
         pluginSetting.setPipelineName(TEST_PIPELINE);
         pluginSetting.setProcessWorkers(TEST_WORKERS);
-
         assertThat(pluginSetting, notNullValue());
         assertThat(pluginSetting.getName(), is(TEST_PLUGIN_NAME));
-        assertThat(pluginSetting.getSettings(), nullValue());
         assertThat(pluginSetting.getPipelineName(), is(TEST_PIPELINE));
         assertThat(pluginSetting.getNumberOfProcessWorkers(), is(TEST_WORKERS));
+    }
+
+    @Test
+    public void testPluginSettingAttributes() {
+        final String TEST_STRING_VALUE = "TEST";
+
+        final Map<String, Object> TEST_SETTINGS = ImmutableMap.of(
+                TEST_INT_ATTRIBUTE, TEST_INT_VALUE,
+                TEST_STRING_ATTRIBUTE, TEST_STRING_VALUE,
+                TEST_BOOL_ATTRIBUTE, TEST_BOOL_VALUE,
+                TEST_LONG_ATTRIBUTE, TEST_LONG_VALUE);
+
+        final PluginSetting pluginSetting = new PluginSetting(TEST_PLUGIN_NAME, TEST_SETTINGS);
+
+        assertThat(pluginSetting.getAttributeFromSettings(TEST_INT_ATTRIBUTE), is(TEST_INT_VALUE));
+
+        // test attributes that exist when passing in a different default value
+        assertThat(pluginSetting.getAttributeOrDefault(TEST_INT_ATTRIBUTE, TEST_INT_DEFAULT_VALUE), is(TEST_INT_VALUE));
+        assertThat(pluginSetting.getIntegerOrDefault(TEST_INT_ATTRIBUTE, TEST_INT_DEFAULT_VALUE), is(TEST_INT_VALUE));
+        assertThat(pluginSetting.getStringOrDefault(TEST_STRING_ATTRIBUTE, TEST_STRING_DEFAULT_VALUE),
+                is(equalTo(TEST_STRING_VALUE)));
+        assertThat(pluginSetting.getBooleanOrDefault(TEST_BOOL_ATTRIBUTE, TEST_BOOL_DEFAULT_VALUE), is(equalTo(TEST_BOOL_VALUE)));
+        assertThat(pluginSetting.getLongOrDefault(TEST_LONG_ATTRIBUTE, TEST_LONG_DEFAULT_VALUE), is(equalTo(TEST_LONG_VALUE)));
```

Comment:
nit - Personal preference would be to split these larger tests up into multiple tests per type (string/long/bool/int), but seeing as we don't expect this class to change much it works

Reply:
i agree with that structure, will do that